test/testdata/parser/kwnil.rb

```diff
@@ -0,0 +1,6 @@
+# typed: true
+
+def f1(**nil); end
+def f2(*args, **nil); end
+def f3(a, b, c, **nil); end
+def f4(a, b, *c, **nil); end
```

Comment:
In the grammar file it looks like there's some logic to handle an optional block arg after the `**nil` construct?

Would you mind writing a test to cover that case? like

```ruby
def f5(**nil, &blk); end
```

Even if it's a syntax error it's good to record it.

I'd also appreciate adding a similar case to your `infer/` test as well if it's valid syntax.

Reply:
See https://github.com/sorbet/sorbet/pull/3412/files#r486369225